src/App.web.tsx

```diff
@@ -26,6 +29,7 @@ const App = observer(function AppImpl() {
       setRootStore(store)
       analytics.init(store)
     })
+    dynamicActivate(defaultLocale) // async import of locale data
```

Comment:
This looks concerning to me. What will it be doing while the locale data is loading? Is this blocking? If it's not blocking, how does the UI work in the meantime?

Reply:
You can check the `i18n.ts` code -- it's just a dynamic import

Reply:
I understand it's a dynamic import — what I don't understand is what strings will we be showing until the locale loads?

Reply:
We can either show nothing or the default locale (English)

Reply:
Let's see if any screen flashing becomes an issue...

Reply:
Things to verify on mobile:

- we shouldn't flash the wrong locale
- we shouldn't introduce any new async delays; all files are already on the disk

Things to verify on web:

- we shouldn't flash the wrong locale
- we shouldn't introduce a waterfall (fetch code -> try to render -> discover we don't have a locale -> download locale). if locale is known at the request time, it should download in parallel with the code.


At FB I _think_ we'd just create a complete JS bundle per locale and force a reload on locale change. I wonder if an approach like this would work for us? 

Also, its docs are not great, but I strongly recommend investigating how https://facebook.github.io/fbt/ handles this.

Reply:
Lingui works very similar to FBT from what I can tell. I'll take a deeper look and then we can adjust the method that we use. However, I do think that *not* reloading the app on locale change would be a better user experience.

Reply:
On mobile, everything will be bundled automatically and then switch when we want it to. I expect our locale `messages.js` files to be around `20kb`. Right now its `14kb` for English.

On web, it will be fetched when required. We can cache it and that will definitely help.  Here is how it works on web: https://lingui.dev/guides/dynamic-loading-catalogs#conclusion

On RN, we would have to implement our own solution. There has been some prior art on this: https://github.com/lingui/js-lingui/issues/503